src/PrestaShopBundle/Form/Admin/Improve/International/Locations/StateType.php

```diff
@@ -0,0 +1,174 @@
+                        'message' => $this->translator->trans(
+                            'This field cannot be empty.', [], 'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new Length([
+                        'max' => StateSettings::MAX_NAME_LENGTH,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters.',
+                            ['%limit%' => 64],
+                            'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new TypedRegex([
+                        'type' => 'generic_name',
+                    ]),
+                ],
+            ])
+            ->add('iso_code', TextType::class, [
+                'label' => $this->translator->trans('ISO code', [], 'Admin.International.Feature'),
+                'help' => $this->translator->trans('1 to 4 letter ISO code.', [], 'Admin.International.Help')
+                    . ' '
+                    . $this->translator->trans('You can prefix it with the country ISO code if needed.', [], 'Admin.International.Help'),
+                'required' => true,
+                'constraints' => [
+                    new NotBlank(),
+                    new Length(
+                        [
+                            'max' => StateSettings::MAX_ISO_CODE_LENGTH,
+                            'maxMessage' => $this->translator->trans(
+                                'This field cannot be longer than %limit% characters.',
+                                ['%limit%' => StateSettings::MAX_ISO_CODE_LENGTH],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]
+                    ),
+                    new TypedRegex([
+                        'type' => 'state_iso_code',
+                    ]),
+                    new CleanHtml(),
+                    new UniqueStateIsoCode([
+                        'excludeId' => $stateIdValue,
+                    ]),
+                ],
+            ])
+            ->add('id_country', ConfigurableCountryChoiceType::class, [
+                'label' => $this->translator->trans('Country', [], 'Admin.Global'),
+                'help' => $this->translator->trans('Country where the state is located.', [], 'Admin.International.Help')
+                    . ' '
+                    . $this->translator->trans('Only the countries with the option "contains states" enabled are displayed.', [], 'Admin.International.Help'),
+                'required' => true,
+                'constraints' => [
+                    new NotBlank(),
+                ],
+                'placeholder' => '--',
+                'choices' => $this->countriesChoiceProvider->getChoices([
+                    'contains_states' => true,
+                ]),
+            ])
+            ->add('id_zone', ZoneChoiceType::class, [
```

Comment:
I know it's already in the legacy form so I won't discuss this But don't you think it's a bit useless to assign the zone to the state?

Since it's already assigned to the country it should be deduced from the country, leaving this editable only increases the risk of inconsistent data (different zone on state and country) But nevermind it's not the purpose of this PR anyway